src/hocon_schema.erl

```diff
@@ -300,8 +300,8 @@ find_structs_per_type(_Schema, _Type, Acc, _Stack, _TStack) ->
     Acc.
 
 find_ref(Schema, Name, Acc, Stack, TStack) ->
-    Namespace = hocon_schema:namespace(Schema),
-    Key = {Namespace, Name},
+    Namespace = namespace(Schema),
+    Key = {Schema, Namespace, Name},
```

Comment:
The lookup key is ok to be only namespace because Schema is not necessary always an atom

Reply:
schema is able to sort even not an atom.